Library/EnvelopeFactory.ts

```diff
@@ -115,7 +115,9 @@ class EnvelopeFactory {
 
     private static createDependencyData(telemetry: Contracts.DependencyTelemetry & Contracts.Identified): Contracts.Data<Contracts.RemoteDependencyData> {
         var remoteDependency = new Contracts.RemoteDependencyData();
-        remoteDependency.name = telemetry.name.length > 1024 ? telemetry.name.slice(0, 1021) + '...' : telemetry.name;
+        if (typeof telemetry.name === "string") {
```

Comment:
When null you will not send a name, is this not causing issues during ingestion?, maybe having default empty string?

Reply:
Name is not required for ingestion. Really the typings are wrong, but they are autogenerated from the bond files and so I would like to avoid creating discrepancies across language SDKs by making a one-off change for our types